clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1269,6 +1283,11 @@ private ConsumerRecords<K, V> poll(final Timer timer, final boolean includeMetad
         }
     }
 
+    private int sendFetches() {
+        offsetFetcher.validatePositionsOnMetadataChange();
```

Comment:
The relocation of this makes me wonder if it's needed at all. We already call the same method in `updateFetchPositions`, which is invoked prior to `sendFetches`. I tried removing it locally and all tests still pass. Probably not a good idea to remove here in the refactor, but maybe we could do it in a follow-up. That would simplify the `OffsetFetcher` api a little.